src/app/services/user.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Observable, lastValueFrom } from 'rxjs';
+import { AuthResponse } from '../models/AuthResponse';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  isLoggedIn: boolean;
```

Comment:
This property is useful, but not in this state. What I would do here is replace `isLoggedIn: boolean` with `user: User | null` and make an observable out of this value. It would look something like this:
```suggestion
  private _userSubject = new BehaviorSubject<User | null>(null); // initial value
  user$ = this._userSubject.asObservable(); // i think
```
Then, when "pushing" a new user to the subject:
```ts
// successful login, etc, user data is in `user`
this._userSubject.next(user);
```
Then, in our components, we can subscribe to this Observable and always have an automatically-updated reference to the current user.
```ts
// observe
userService.user$.subscribe((user) => /* do something with the user, and whenever it changes */);
```